univ_rd/univ_rd.R

```diff
@@ -36,12 +36,14 @@ county_univRD <- NSF_univRD %>%
   select(-county)
 
 
-labels<-c("total R&D spending",
-          "total deflated business financed R&D spending",
-          "FIPS code")
+labels<-c(rd_total_biz = "total deflated business financed R&D spending",
+          rd_total =  "total R&D spending",
+          stco_code = "FIPS code",
+          cbsa_code = "FIPS code")
+
+label(county_univRD) = as.list(labels[match(names(county_univRD), names(labels))])
```

Comment:
This line of code assigns the "labels" vector to the dataframe in a soft-coded manner. The vector's elements are paired (column name = label) and the "match" feature ensures that the label is assigned to the corresponding column name in the dataset: *even if the columns in the dataset are in a different order or if some columns are absent, no mislabeling will occur.